src/main/resources/assets/carpet/scripts/shapes.scl

```diff
@@ -97,134 +132,178 @@ draw_prism(args)->(
     positions
 );
 
-fill_flat(pos, offset, dr, rectangle, orientation, hollow)->
+//Allows to fill a rectangle or circle
+//'pyramid' parameter is cos this function is called by draw_prism and draw_pyramid
+//Now in pyramids, I want circles to be ever so slightly thicker,
+//whereas in prisms I don't, hence the parameter.
+//The reason I do abs() of the radius for _fill_flat_circle is because sometimes,
+//the calculations result in -0 being inputted as a radius.
```

Comment:
Another (probably cheaper?) way to get rid of `-0` should just be adding 0 to it. Wondering, how does it cause problems? It should behave like 0 in most cases.

Reply:
Well the reason I opened #1374 was because it was breaking this. It was actually quite a hassle to figure out the culprit, but my workaround adds basically 0 compute.